### PR TITLE
Convert query to use standard dask query and update GPUCI to cudf 0.7

### LINKED ISSUE #128: [FEA] Support local variables in query API
## Description
As a user, I will want to pass local variables to query dask_cudf DataFrames. This is particularly useful for timestamp and datetime based querying. Am I able to do this with the `local_dict` keyword argument in standard dask, but it's currently not implemented in dask_cudf.

## Standard Dask Example
```
import cudf
import numpy as np
import pandas as pd
import dask_cudf
import dask.dataframe as dd
import datetime as dt

np.random.seed(12)

date_df = cudf.DataFrame()
date_df['date'] = pd.date_range('11/20/2018', periods=72, freq='D')
date_df['value'] = np.random.sample(len(date_df))

pdf = date_df.to_pandas()
pddf = dd.from_pandas(pdf, npartitions=2)

search_date = dt.datetime.strptime('2018-11-23', '%Y-%m-%d')

pddf.query('date <= @search_date', local_dict={'search_date':search_date}).compute()
        date     value
0 2018-11-20  0.154163
1 2018-11-21  0.740050
2 2018-11-22  0.263315
3 2018-11-23  0.533739
```

## Dask Cudf Example
Passing a `local_dict` results in a keyword argument error:
```
import cudf
import numpy as np
import pandas as pd
import dask_cudf
import dask.dataframe as dd
import datetime as dt

np.random.seed(12)
date_df = cudf.DataFrame()
date_df['date'] = pd.date_range('11/20/2018', periods=72, freq='D')
date_df['value'] = np.random.sample(len(date_df))

date_ddf = dask_cudf.from_cudf(date_df, npartitions=2)

search_date = dt.datetime.strptime('2018-11-23', '%Y-%m-%d')

print(date_ddf.query('date <= @search_date', local_dict={'search_date':search_date}).compute())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-22-7a40ec2e6f3a> in <module>
     15 search_date = dt.datetime.strptime('2018-11-23', '%Y-%m-%d')
     16 
---> 17 print(date_ddf.query('date <= @search_date', local_dict={'search_date':search_date}).compute())

TypeError: query() got an unexpected keyword argument 'local_dict'
```

But, not passing a `local_dict` gives a local variable not implemented exception.
```
import cudf
import numpy as np
import pandas as pd
import dask_cudf
import dask.dataframe as dd
import datetime as dt

np.random.seed(12)
date_df = cudf.DataFrame()
date_df['date'] = pd.date_range('11/20/2018', periods=72, freq='D')
date_df['value'] = np.random.sample(len(date_df))

date_ddf = dask_cudf.from_cudf(date_df, npartitions=2)
search_date = dt.datetime.strptime('2018-11-23', '%Y-%m-%d')

print(date_ddf.query('date <= @search_date'))
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-12-fee85ff69cd6> in <module>
     14 search_date = dt.datetime.strptime('2018-11-23', '%Y-%m-%d')
     15 
---> 16 print(date_ddf.query('date <= @search_date'))

/conda/envs/cudf_dev/lib/python3.7/site-packages/dask_cudf-0.0.1+252.g0f86985-py3.7.egg/dask_cudf/core.py in query(self, expr)
    237         """
    238         if "@" in expr:
--> 239             raise NotImplementedError("Using variables from the calling " "environment")
    240         # Empty calling environment
    241         callenv = {"locals": {}, "globals": {}}

NotImplementedError: Using variables from the calling environment
```

### PR DESCRIPTION
**Pull request by beckernick**
**Summary of Changes**
- Remove dask_cudf specific query functionality in favor of standard dask query implementation
- Add additional query unit test for handling local dictionary values
- Update GPUCI to use cudf 0.7 and enforce package version requirements

This PR depends on cuDF 0.7. As a result it also updates the GPUCI to cudf 0.7 and resolves a bug that caused prior versions of GPUCI for dask-cudf to not enforce package version requirements.

This closes #128 and also closes PR #206 (206 is subsumed by this PR).

### CODE REVIEW

**Comment by mrocklin** [comment]
I'm fine relaxing the requirements constraint

**Comment by mrocklin** [comment]
Does anyone know the right way to rerun tests?

**Comment by beckernick** [comment]
> Does anyone know the right way to rerun tests?

I know you figured it out but its the "rerun tests" in that sentence 👍

**Comment by mrocklin** [comment]
Ah, good to know.  Thanks @beckernick !

**Comment by raydouglass** [comment]
rerun tests

**Comment by mrocklin** [comment]
It looks like we're getting failures because `local_dict` isn't in the installed version of cudf.  It looks like this was added to cudf recently, Lets try rerun tests again.

**Comment by mrocklin** [comment]
rerun tests

**Comment by mrocklin** [comment]
My attempts to rerun tests seem to be ineffective.  I'm not sure if there is another way I can do this.

**Comment by mrocklin** [comment]
Barring any more clever solution you might try pushing a trivial change to this PR @beckernick .  I wish I knew of a nicer solution for you :/

**Comment by kkraus14** [comment]
> Barring any more clever solution you might try pushing a trivial change to this PR @beckernick . I wish I knew of a nicer solution for you :/

Pushing any commit (even an empty commit) should resolve the issues. This was due to a gpuCI rollover and github won't clear the statuses unless there's a new commit.

**Comment by beckernick** [comment]
Thanks Keith. Will do

**Comment by mrocklin** [comment]
Still getting this error, which is interesting:

```python
E           TypeError("query() got an unexpected keyword argument 'local_dict'",)
```

**Comment by beckernick** [comment]
@mrocklin Tests are still failing because CI is tied to cudf 0.6. Are we able to update this to 0.7 ourselves or do we need @raydouglass or @mike-wendt to make that updat?

**Comment by mrocklin** [comment]
It looks like we might be able to change things here:

https://github.com/rapidsai/dask-cudf/blob/3279ae6cc07e38eb26f67d639b4fae4c401e5d7e/ci/gpu/build.sh#L39-L44

**Comment by mrocklin** [comment]
Want to give it a try and see how things go?

**Comment by mrocklin** [comment]
I would expect that using the nightly conda channel would solve this

**Comment by beckernick** [comment]
Good point. Will give that a test.

**Comment by beckernick** [comment]
rerun tests (updated CI environment)

**Comment by mrocklin** [comment]
Looks like we're still installing 0.6.1 here

    cudf                      0.6.1                    py36_0    rapidsai/label/cuda10.0

**Comment by beckernick** [comment]
Passing this test is blocked by #206 , @mrocklin

**Comment by beckernick** [comment]
Bringing in the GPUCI updates to this PR to be able to pass all at once. Not that elegant and will affect the history, but I'd rather confirm it all passes.

**Comment by beckernick** [comment]
@mrocklin , tests passing. GPUCI is also now [correctly enforcing version requirements](https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/gpuCI%2Fdask-cudf%2Fprb%2Fdask-cudf-gpu-build/detail/dask-cudf-gpu-build/30/pipeline#log-152).

File: dask_cudf/tests/test_core.py [hunk 1]
```diff
@@ -74,6 +74,23 @@ def test_query():
...
+        {"x": np.random.randint(0, 5, size=10), "y": np.random.normal(size=10)}
+    )
+    gdf = cudf.DataFrame.from_pandas(df)
+    ddf = dgd.from_cudf(gdf, npartitions=2)
+
+    val = 2
+
+    gdf_queried = gdf.query("x > @val").to_pandas()
+    ddf_queried = ddf.query("x > @val",
+                            local_dict={'val': val}).compute().to_pandas()
```
Comment (mrocklin) [inline dask_cudf/tests/test_core.py]:
```suggestion
                            local_dict={'val': val})
```

File: dask_cudf/tests/test_core.py [hunk 2]
```diff
@@ -74,6 +74,23 @@ def test_query():
...
+    np.random.seed(0)
+    df = pd.DataFrame(
+        {"x": np.random.randint(0, 5, size=10), "y": np.random.normal(size=10)}
+    )
+    gdf = cudf.DataFrame.from_pandas(df)
+    ddf = dgd.from_cudf(gdf, npartitions=2)
+
+    val = 2
+
+    gdf_queried = gdf.query("x > @val").to_pandas()
```
Comment (mrocklin) [inline dask_cudf/tests/test_core.py]:
```suggestion
    gdf_queried = gdf.query("x > @val")
```

File: dask_cudf/tests/test_core.py (hunk 1 above)
Comment (beckernick) [reply dask_cudf/tests/test_core.py]: Noted. Will update shortly.

Comment (mrocklin) [reply dask_cudf/tests/test_core.py]: You should be able to press the "Commit suggestion" button above

Comment (beckernick) [reply dask_cudf/tests/test_core.py]: Well that was fancy!

Comment (mrocklin) [reply dask_cudf/tests/test_core.py]: :)

**Comment by beckernick** [comment]
Downside of using the fancy commit from suggestion is that you can't pytest locally 😄 . Will address the test failure.

**Comment by beckernick** [comment]
@mrocklin the `assert_frame_equal` functionality that was already being used in other tests does require pandas objects. I can switch to use dask's `assert_eq`, but for consistency I'm going to revert back to the prior commit.

**Comment by mrocklin** [comment]
I just re-reverted your changes, and also changed all of the other cases to use `dd.assert_eq` (there is a bit of old code around that we're rewriting over time).

Merging after tests pass

**Pull request merged**